src/BlazorWebView/src/core/iOS/BlazorWebViewHandler.iOS.cs

```diff
@@ -153,10 +177,28 @@ public static void MapRootComponents(BlazorWebViewHandler handler, IBlazorWebVie
 			handler.StartWebViewCoreIfPossible();
 		}
 
-		public static void MapServices(BlazorWebViewHandler handler, IBlazorWebView webView)
+		private void OnRootComponentsCollectionChanged(object? sender, global::System.Collections.Specialized.NotifyCollectionChangedEventArgs eventArgs)
 		{
-			handler.Services = webView.Services;
-			handler.StartWebViewCoreIfPossible();
+			// If we haven't initialized yet, this is a no-op
+			if (_webviewManager != null)
+			{
+				// Dispatch because this is going to be async, and we want to catch any errors
+				_ = _webviewManager.Dispatcher.InvokeAsync(async () =>
+				{
+					var newItems = eventArgs.NewItems!.Cast<RootComponent>();
+					var oldItems = eventArgs.OldItems!.Cast<RootComponent>();
+
+					foreach (var item in newItems.Except(oldItems))
```

Comment:
Can this be a problem here? I'm not sure if we've defined equality for the root component type.

Reply:
I would have thought the default struct equality rules would have the right behavior in this case.

Reply:
It's the same exact code from WPF/WinForms. I'll log an issue to confirm.